cve_bin_tool/helper_script.py

```diff
@@ -26,14 +23,19 @@ class HelperScript:
     """Helps contributors who want to write a new cve-bin-tool checker find common filenames, version strings, and other necessary data for building a binary checker"""
 
     CONSOLE = Console()
+    LOGGER = LOGGER.getChild("HelperScript")
 
-    def __init__(self, filename, product_name=None):
+    def __init__(
+        self, filename, product_name=None, version_number=None, string_length=40
+    ):
         self.filename = filename
```

Comment:
```
def __init__(
        self, filename, product_name=None, version_number=None, string_length=40
    ):
        self.filename = filename
        self.extractor = Extractor()
        self.product, self.version = self.parse_filename(filename)
```
filename is used here to parse itself and it's also used now at some places in `logs`.

Reply:
I mean `self.filename`. It is not used anywhere (I just searched `.filename`).

Thanks, now I see that you actually get `self.product, self.version` from this `filename` argument. But why is it required? If someone passes both `product_name` and `version_number`, you will use them and discard what you parsed from `filename`, right? This is what happens here:
https://github.com/intel/cve-bin-tool/blob/45578e8c9ab79ae872ee432692901e812431f7fc/cve_bin_tool/helper_script.py#L400-L405

And another note: you use `filename/filename, string_length/string_length` for init argument/class arguments but `product_name/product, version_number/version`. I'd suggest making them the same names for consistency.

Reply:
> And another note: you use filename/filename, string_length/string_length for init argument/class arguments but product_name/product, version_number/version. I'd suggest making them the same names for consistency.

I'll make these changes now :)

> Thanks, now I see that you actually get self.product, self.version from this filename argument. But why is it required? If someone passes both product_name and version_number, you will use them and discard what you parsed from filename, right?

I also do not quite like this here, because if the user specifies the `product_name` and `version_name`, we would discard the above work done. But, since we only scan it against a short string and a few cases, so it should not affect the performance by much and I decided to settle with this.